src/lib/mdns/platform/Mdns.h

```diff
@@ -145,25 +156,16 @@ CHIP_ERROR ChipMdnsShutdown();
 CHIP_ERROR ChipMdnsPublishService(const MdnsService * service);
 
 /**
- * This function stops publishing all services via mDNS.
- *
- * @retval CHIP_NO_ERROR                The publish stops successfully.
- * @retval Error code                   Stopping the publish fails.
+ * Completes updating advertised services.
  *
- */
-CHIP_ERROR ChipMdnsStopPublish();
-
-/**
- * This function stops publishing a specific service via mDNS.
- *
- * @param[in] service   The service entry.
- *
- * @retval CHIP_NO_ERROR                The service stop succeeds.
- * @retval CHIP_ERROR_INVALID_ARGUMENT  The service is nullptr.
- * @retval Error code                   The service stop fails.
+ * This function can be used by some implementations to apply changes made with the
+ * `ChipMdnsRemoveServices` and `ChipMdnsPublishService` functions in case they could
+ * not be applied immediately (like in case of implementations using SRP).
```

Comment:
I think we should assume asynchronous in every case, rather than assume just SRP requires asynchrony.

Reply:
SRP is here provided just as an example, not the only case where it's needed. Current Avahi implementation has the same issue as Avahi doesn't allow to modify entry groups which have been committed, so currently Linux lighting-app built with Avahi support can't correctly advertise multiple services (I'm going to file a follow up for that and use `CompleteServiceUpdate` to fix that).

Reply:
> Current Avahi implementation has the same issue as Avahi doesn't allow to modify entry groups 

This is a great reason to finally stop using the singleton pattern here.  Can't avahi support many clients on the dbus interface?  If we separated each usage into its own instance / client, we could avoid all of this undesirable interaction.

Reply:
It is possible to have mutliple Avahi clients, but it's a bit different issue in my opinion. Avahi uses a concept called "entry group" which is basically a handle to a group of services and related data. Once an entry group is commited with `avahi_entry_group_commit` function, the services associated with the entry group cannot be modified (except the TXT entries) without resetting the whole entry group. The current `Linux/MdnsImpl.cpp` implementation uses a single entry group for all services and the entry group is commited in `Advertise` call so any subsequent `Advertise` calls fail. Therefore, we should move the `avahi_entry_group_commit` to `CompleteServiceUpdate` or use multiple entry groups.